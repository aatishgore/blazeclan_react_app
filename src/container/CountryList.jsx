import Container from "../components/Container";
import { getCountry } from "../services/services";
import React, { useState, useEffect } from 'react';
import { errorHandler } from "../lib/utils/helper";

const CountryList= () => {
    // Declare state
    const [serachValue, setSearchValue] = useState(null);
    const [pagination, setPagination] = useState(false);
    const [countryList, setCountryList] = useState([]);
    const [loading, setLoading] = useState(false);
    const [error, setError] = useState(null);
    const [limit, setLimit] = useState(10);
    const [page, setPage] = useState(1);
    
    // Call API here to get country
    useEffect(() => {
        getCountryStarted();
        getCountry({ page, limit, serachValue }).then((res) => {
            getCountrySuccess(res.data);
        }).catch((err) => {
            getCountryFail(errorHandler(err));
        })
    }, [page, limit, serachValue]);

    // Here we are set state as per the api response
    const getCountryStarted = () => {
        setLoading(true);
    };
    const getCountrySuccess = ({data, pagination}) => {
        setError(null);
        setCountryList(data);
        setPagination(pagination);
        setLoading(false);
    };
    const getCountryFail = err => {
        setError(err);
        setCountryList([]);
        setPagination(false);
        setLoading(false);
    };

    return (
        <Container
            error={error}
            loading={loading}
            pagination={pagination}
            countryList={countryList}
            onClose={()=> setError(null)}
            handleOnChange={(value)=> setSearchValue(value)}
            handleOnChangePage={(pageNumber,pageSize) => {
                setPage(pageNumber)
                setLimit(pageSize)
            }}
            handleOnSizeChange={(pageNumber,pageSize) => {
                setPage(pageNumber)
                setLimit(pageSize)
            }}
        />
    )
}
export default CountryList;