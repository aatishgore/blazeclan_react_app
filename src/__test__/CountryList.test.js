import CountryList from "../container/CountryList";
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import React from 'react';

configure({ adapter: new Adapter() });

describe('CountryList', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<CountryList debug />);
        expect(component).toMatchSnapshot();
    });
});