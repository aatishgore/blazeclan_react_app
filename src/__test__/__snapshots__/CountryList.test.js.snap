// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryList should render correctly in "debug" mode 1`] = `
<Container
  countryList={Array []}
  error={null}
  handleOnChange={[Function]}
  handleOnChangePage={[Function]}
  handleOnSizeChange={[Function]}
  loading={false}
  onClose={[Function]}
  pagination={false}
/>
`;
